pkg/mic/mic.go

```diff
@@ -237,6 +238,11 @@ func (c *Client) Sync(exit <-chan struct{}) {
 		begin := time.Now()
 		workDone := false
 
+		cacheTime := time.Now()
+		// sync cache to get the latest data
```

Comment:
Please elaborate the comment here something which says the following - "since we could have done creates in the previous cycle and which has not yet been propagated we might end up doing the cycles again. Syncing the cache avoids such situations. "